examples/gno.land/r/demo/hof/rendering.gno

```diff
@@ -0,0 +1,117 @@
+package hof
+
+import (
+	"strings"
+
+	"gno.land/p/demo/avl/pager"
+	"gno.land/p/demo/fqname"
+	"gno.land/p/demo/seqid"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/moul/txlink"
+)
+
+const (
+	pageSize = 5
+)
+
+func Render(path string) string {
+	out := "# Hall of Fame\n\n"
+
+	dashboardEnabled := path == "dashboard"
+
+	if dashboardEnabled {
+		out += renderDashboard()
+	}
+
+	out += exhibition.Render(path, dashboardEnabled)
+
+	return out
+}
+
+func (e Exhibition) Render(path string, dashboard bool) string {
+	out := ufmt.Sprintf("%s\n\n", e.description)
+
+	if e.items.Size() == 0 {
+		out += "No items in this exhibition currently.\n\n"
+		return out
+	}
+
+	out += "<div class='columns-2'>\n\n"
+
+	page := pager.NewPager(e.itemsSorted, pageSize).MustGetPageByPath(path)
+
+	for i := len(page.Items) - 1; i >= 0; i-- {
+		item := page.Items[i]
+
+		out += "<div>\n\n"
+		id, _ := seqid.FromString(item.Key)
+		out += ufmt.Sprintf("### Submission #%d\n\n", int(id))
+		out += item.Value.(*Item).Render(dashboard)
+		out += "</div>"
+	}
+
+	out += "</div><!-- /columns-2 -->\n\n"
+
+	out += page.Selector()
+
+	return out
+}
+
+func (i Item) Render(dashboard bool) string {
+	out := ufmt.Sprintf("\n```\n%s\n```\n\n", i.pkgpath)
+	out += ufmt.Sprintf("by %s\n\n", strings.Split(i.pkgpath, "/")[2])
+	out += ufmt.Sprintf("[View realm](%s)\n\n", fqname.ShortPath(i.pkgpath))
+	out += ufmt.Sprintf("Submitted at Block #%d\n\n", i.blockNum)
+
+	out += ufmt.Sprintf("#### [%d👍](%s) - [%d👎](%s)\n\n",
+		i.upvote.Size(), txlink.URL("Upvote", "pkgpath", i.pkgpath),
+		i.downvote.Size(), txlink.URL("Downvote", "pkgpath", i.pkgpath),
+	)
+
+	if dashboard {
+		out += ufmt.Sprintf("[Delete](%s)", txlink.URL("Delete", "pkgpath", i.pkgpath))
+	}
+
+	return out
+}
+
+func renderDashboard() string {
+	out := "---\n\n"
+	out += "## Dashboard\n\n"
+	out += ufmt.Sprintf("Total submissions: %d\n\n", exhibition.items.Size())
+
+	out += ufmt.Sprintf("Exhibition admin: %s\n\n", owner.Owner().String())
+
+	if !exhibition.IsPaused() {
+		out += ufmt.Sprintf("[Pause exhibition](%s)\n\n", txlink.URL("Pause"))
+	} else {
+		out += ufmt.Sprintf("[Unpause exhibition](%s)\n\n", txlink.URL("Unpause"))
+	}
+
+	out += "---\n\n"
+
+	return out
+}
+
+func RenderExhibWidget(itemsToRender int) string {
+	if itemsToRender < 1 {
+		return ""
+	}
+
+	if exhibition.items.Size() == 0 {
+		return ""
+	}
```

Comment:
this is not needed. If there are no items, the output will be "" anyway.

Reply:
You're right; rm'd the condition: [ff06493](https://github.com/gnolang/gno/pull/2842/commits/ff06493e12c7dae579ea2c0cbcb9d3eee276f4f7)